### PR TITLE
[Rule Tuning] Access to Keychain Credentials Directories

### PR DESCRIPTION
**Pull request by Samirbous**
## Issues

Resolves https://github.com/elastic/detection-rules/issues/997

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/macos/credential_access_credentials_keychains.toml [hunk 1]
```diff
@@ -35,7 +35,18 @@ process where event.type in ("start", "process_started") and
...
+                      "add-trusted-cert",
+                      "set-keychain-settings",
+                      "delete-certificate",
+                      "/Users/*/Library/Keychains/openvpn.keychain-db",
+                      "show-keychain-info",
+                      "lock-keychain",
+                      "set-key-partition-list",
+                      "import",
+                      "find-identity") and
+    not process.parent.executable:"/Applications/OpenVPN Connect/OpenVPN Connect.app/Contents/MacOS/OpenVPN Connect"
```
Comment (brokensound77) [inline rules/macos/credential_access_credentials_keychains.toml:49]: nit:

```suggestion
    ) and
    not process.args : ("find-certificate",
                      "add-trusted-cert",
                      "set-keychain-settings",
                      "delete-certificate",
                      "/Users/*/Library/Keychains/openvpn.keychain-db",
                      "show-keychain-info",
                      "lock-keychain",
                      "set-key-partition-list",
                      "import",
                      "find-identity") and
    not process.parent.executable : "/Applications/OpenVPN Connect/OpenVPN Connect.app/Contents/MacOS/OpenVPN Connect"
```

**Review by brokensound77 — APPROVED** [review]
bump `updated_date` and LGTM 👍

File: rules/macos/credential_access_credentials_keychains.toml [hunk 2]
```diff
@@ -36,7 +36,18 @@ process where event.type in ("start", "process_started") and
...
+                      "add-trusted-cert",
+                      "set-keychain-settings",
+                      "delete-certificate",
+                      "/Users/*/Library/Keychains/openvpn.keychain-db",
+                      "show-keychain-info",
+                      "lock-keychain",
+                      "set-key-partition-list",
+                      "import",
+                      "find-identity") and
+    not process.parent.executable : "/Applications/OpenVPN Connect/OpenVPN Connect.app/Contents/MacOS/OpenVPN Connect"
```
Comment (bm11100) [inline rules/macos/credential_access_credentials_keychains.toml:50]: would you consider eliminating the root user as well? bring the noise down a bit more.

Comment (Samirbous) [reply rules/macos/credential_access_credentials_keychains.toml:50]: access to those folders requires root, if filtered-out may risk to miss stuff.

Comment (bm11100) [reply rules/macos/credential_access_credentials_keychains.toml:50]: makes sense, i wasn't sure if that was the case. i def saw non root users trigger this as well. but i think the numbers are low enough.

**Review by bm11100 — APPROVED** [review]
🍟

**Pull request merged**